### PR TITLE
Update tests to take advantage of cats-effect-0.10

### PR DESCRIPTION
**Pull request by aeons**
Test part of #1736

### CODE REVIEW

File: server/src/test/scala/org/http4s/server/middleware/TimeoutSpec.scala [hunk 1]
```diff
@@ -45,22 +46,20 @@ class TimeoutSpec extends Http4sSpec {
       checkStatus(altTimeoutService.orNotFound(neverReq), customTimeout.status)
     }
 
-    "clean up resources of the loser" in {
-      val clean = new AtomicBoolean(false)
+    "cancel the loser" in {
+      val cancelled = new AtomicBoolean(false)
```
Comment (rossabaker) [inline server/src/test/scala/org/http4s/server/middleware/TimeoutSpec.scala]: English uses both, but cats-effect uses "canceled".

It's not part of the API, so I won't die on this hill.

Comment (ChristopherDavenport) [reply server/src/test/scala/org/http4s/server/middleware/TimeoutSpec.scala]: Nitpicky much...

File: server/src/test/scala/org/http4s/server/middleware/TimeoutSpec.scala [hunk 2]
```diff
@@ -47,19 +47,19 @@ class TimeoutSpec extends Http4sSpec {
     }
 
     "cancel the loser" in {
-      val cancelled = new AtomicBoolean(false)
+      val canceled = new AtomicBoolean(false)
```
Comment (jmcardon) [inline server/src/test/scala/org/http4s/server/middleware/TimeoutSpec.scala]: The travis failure is deserve. Queens english ftw.

**Pull request merged**